e2e_tests/tests/command/test_run.py

```diff
@@ -398,7 +398,11 @@ def test_cmd_kill() -> None:
         assert command.task_id is not None
         for line in command.stdout:
             if "hello world" in line:
-                assert cmd.get_num_running_commands() == 1
+                # For HPC job, dispatcher does the polling of the job state happens
+                # every 10 seconds. For example, it is very likely the current job state is
+                # STATE_PULLING when job is actually running on HPC. So instead of checking
```

Comment:
But the Determined command is not actually running if the image is being pulled.  Wouldn't you want to wait for the image to get pulled and the command to run before terminating the command?

Reply:
> But the Determined command is not actually running if the image is being pulled. Wouldn't you want to wait for the image to get pulled and the command to run before terminating the command?

The command is running when "hello world" is printed in stdout. Determined has not got the most recent slurm job state. Dispatch monitoring polls the job state every 10 seconds. The alternative is to wait for 10 seconds after seeing the "hello world" line

```
    with cmd.interactive_command(
        "command", "run", "echo hello world; echo hello world; sleep infinity"
    ) as command:
        assert command.task_id is not None
        for line in command.stdout:
            if "hello world" in line:
```

Reply:
Oh, I see.  Thank you.